### PR TITLE
feat(deprecation): add ability to deprecate packages

### LINKED ISSUE #13: Add support for deprecating packages
There are some scenarios where we may want to deprecate already published packages, for example #11.

### PR DESCRIPTION
**Pull request by corymhall**
This introduces the ability to deprecate packages.

When a user adds a package to the list of `deprecatedTypes` (projenrc/deprecated-types.ts), it will

1. Add a disclaimer to the package README that the package is deprecated
2. Stop taking registry updates for the package
3. Add `deprecated: true` to the package.json

fixes #13

### CODE REVIEW

File: projenrc/type-package.ts [hunk 1]
```diff
@@ -163,88 +180,130 @@ export class CloudFormationTypeProject extends Component {
...
+          },
+        },
+        name: `deprecate-${typeNameKebab}`,
+        task: deprecateTask,
+        env: {
+          NPM_REGISTRY: 'registry.npmjs.org',
+          NPM_CONFIG_REGISTRY: 'https://registry.npmjs.org',
         },
+        preBuildSteps: [
+          { run: 'echo "//${{ env.NPM_REGISTRY }}/:_authToken=${{ secrets.NPM_TOKEN }}" > ~/.npmrc' },
```
Comment (corymhall) [inline projenrc/type-package.ts:204]: I'm not sure this is the correct way to do this...

**Review by eladb — CHANGES_REQUESTED** [review]
If we remove the release workflows for the deprecated packages, then non-npm package managers won't see the deprecation. I am wondering if the right thing to do is toL

* Mark it as `deprecated` in `package.json`
* Announce that it is deprecated in `README.md`
* Stop taking registry updates for it
* Continue to publish to all package managers.

We can then decide if we want to actually deprecate it "officially" or just leave it like that.

Maybe we can also special-case this in the Construct Hub so that these will be filtered out even if they are not officially marked as deprecated.

Thoughts?

**Comment by corymhall** [comment]
Yeah I think that makes the most sense. We can always add the package to `excludeTypes` if we want to stop publishing at any point.

Does the `deprecated` property in `package.json` have any function, or is it just for documentation? If this is part of the info that Construct Hub picks up then maybe they can use this to filter.

**Comment by eladb** [comment]
I don't believe it has any function (see https://docs.npmjs.com/cli/v8/configuring-npm/package-json)

**Comment by eladb** [comment]
@corymhall ping

**Comment by corymhall** [comment]
I should be able to get to this in the next couple of days.

**Review by eladb — APPROVED** [review]
GREAT JOB!

**Pull request merged**